src/test/ui/const-generics/generic_const_exprs/const_eval_resolve_canonical.stderr

```diff
@@ -2,13 +2,18 @@ error[E0282]: type annotations needed
   --> $DIR/const_eval_resolve_canonical.rs:26:9
    |
 LL |     let mut _q = Default::default();
-   |         ^^^^^^ consider giving `_q` a type
+   |         ^^^^^^
+   |
+help: consider giving `_q` an explicit type, where the placeholders `_` are specified
```

Comment:
could you specialize the error message to use singular placeholder when there is only a single placeholder?

Reply:
Yes, that'd be lovely, maybe something like "consider giving `_q` an explicit type" and the suggestion be `: Type`.

Reply:
for `let x: _` this now just uses "consider giving `_q` an explicit type". Added a FIXME to special-case types with only a placeholder as that isn't as trivial